UI/src/model/locationSearch.js

```diff
@@ -29,92 +29,95 @@
       } else if (selectedLayer === 'linkProperty') {
-        return roadNumberAndRoadLinkSearch(input);
+        return roadLinkSearchById(input);
       } else if (selectedLayer === 'speedLimit') {
-        return roadNumberAndSpeedLimitSearch(input.text);
+        return speedLimitSearchById(input);
       } else {
-        return getCoordinatesFromRoadAddress({roadNumber: input.text});
+        return $.Deferred().reject('Haulla ei löytynyt tuloksia');
       }
     };
 
     /**
-     * Speed limit id search, combined with road number search
+     * Speed limit id search
      *
      * @param input
      * @returns {*}
      */
-    var roadNumberAndSpeedLimitSearch = function(input){
-      var roadNumberSearch = backend.getCoordinatesFromRoadAddress(input);
-      var speedlimitSearch= backend.getSpeedLimitsLinkIDFromSegmentID(input);
-      return $.when(
-        speedlimitSearch, roadNumberSearch).then(function(speedlimitdata,roadData) {
-        var returnObject = roadLocationAPIResultParser(roadData);
-        if (_.get(speedlimitdata[0], 'success')) {
-          var linkid = _.get(speedlimitdata[0], 'linkId');
-          var y = _.get(speedlimitdata[0], 'latitude');
-          var x= _.get(speedlimitdata[0], 'longitude');
-          var title = input + " (nopeusrajoituksen ID)";
+    var speedLimitSearchById = function(input){
+      return $.when(backend.getSpeedLimitsLinkIDFromSegmentID(input.text)).then(function(speedlimitdata) {
+        var returnObject = [];
+        if (_.get(speedlimitdata, 'success')) {
+          var linkid = _.get(speedlimitdata, 'linkId');
+          var y = _.get(speedlimitdata, 'latitude');
+          var x= _.get(speedlimitdata, 'longitude');
+          var title = input.text + " (nopeusrajoituksen ID)";
             returnObject.push({title: title, lon: x, lat: y, linkid:linkid, resultType:"SpeedLimit"});
         }
-        if (returnObject.length===0){
+
+        if (_.isEmpty(returnObject))
           return $.Deferred().reject('Haulla ei löytynyt tuloksia');
-        }
         return returnObject;
         });
     };
 
     /**
-     * Link id search, combined with road number search
+     * Link id search
      *
      * @param input
      * @returns {*}
      */
-    var roadNumberAndRoadLinkSearch= function(input) {
-      var roadLinkSearch = backend.getRoadLinkToPromise(input.text);
-      var roadNumberSearch = backend.getCoordinatesFromRoadAddress(input.text);
-      return $.when(roadLinkSearch, roadNumberSearch).then(function(linkdata,roadData) {
-        var returnObject = roadLocationAPIResultParser(roadData);
-        if (_.get(linkdata[0], 'success')) {
-          var x = _.get(linkdata[0], 'middlePoint.x');
-          var y = _.get(linkdata[0], 'middlePoint.y');
+    var roadLinkSearchById= function(input) {
+      return $.when(backend.getRoadLinkToPromise(input.text)).then(function(linkdata) {
+        var returnObject = [];
+        if (_.get(linkdata, 'success')) {
```

Comment:
Comparing to previous,  linkdata is an array.
Shouldn't we validate if it is an array?

Reply:
Don´t know why but it was never a array but a object